send-transaction-service/src/transaction_client.rs

```diff
@@ -285,6 +288,60 @@ where
         lock.1.cancel();
         Ok(())
     }
+
+    async fn do_update_key(&self, identity: &Keypair) -> Result<(), Box<dyn std::error::Error>> {
+        let runtime_handle = self.runtime_handle.clone();
+        let config = Self::create_config(
+            Some(identity.insecure_clone()),
+            self.leader_forward_count as usize,
+        );
+        let leader_updater = self.leader_updater.clone();
+        let handle = self.join_and_cancel.clone();
+
+        let join_handle = {
+            let Ok(mut lock) = handle.lock() else {
```

Comment:
nit: let Ok((token, handle)) = ...

Reply:
Sorry but I don't understand what do you propose here

Reply:
My bad you can't destructure there because the guard needs to be alive. But
below you should do:

```
let (handle, token) = lock;
token.cancel();
handle.take();
```

if you use lock.1 and lock.0 I as a reader have no idea what's what.

Reply:
I can take it:

```rust
 let (handle, token) = std::mem::take(&mut *lock);
```